From e4e4366a35b394fd7ef7940651413be206adb5ff Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Mon, 19 Sep 2011 11:40:31 +0200
Subject: fbdev: sh_mobile_meram: Make current_reg field store the current reg
 set

Make sure current_reg == 0/1 always mean register set A/B through all
the code.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
(cherry picked from commit eb4f2304ba029f78516c2fe23213d7e2d0f8d58f)

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 drivers/video/sh_mobile_meram.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/sh_mobile_meram.c b/drivers/video/sh_mobile_meram.c
index 548f700..7af2ffe 100644
--- a/drivers/video/sh_mobile_meram.c
+++ b/drivers/video/sh_mobile_meram.c
@@ -259,8 +259,8 @@ static inline void meram_set_next_addr(struct sh_mobile_meram_priv *priv,
 {
 	unsigned long target;
 
-	target = (cfg->current_reg) ? MExxSARA : MExxSARB;
 	cfg->current_reg ^= 1;
+	target = cfg->current_reg ? MExxSARB : MExxSARA;
 
 	/* set the next address to fetch */
 	meram_write_icb(priv->base, cfg->icb[0].cache_icb,  target,
-- 
1.7.10

